router/batchrouter/batchrouter_test.go

```diff
@@ -299,6 +299,131 @@ var _ = Describe("BatchRouter", func() {
 			cancel()
 			wg.Wait()
 		})
+
+		It("should abort jobs that have retry limits, with lesser(default) limits for rSources jobs", func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
```suggestion
			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
			defer cancel()
```

Reply:
It's being cancelled during the last store call made by the router, `DoAndReturn`.
Also can't defer this cancel, because we're waiting on the `done` channel to close

Reply:
There is no risk of leaking here. If the mock doesn't call `cancel` the timeout triggers and `done` is closed anyway.